### PR TITLE
planner: add a transformation rule FilterAggregateTranspose

### PR DESCRIPTION
**Pull request by zz-jason**
### What problem does this PR solve? 

Trying to add a new transformation rule, which pushes the filter through the aggregate.

This PR reopens https://github.com/pingcap/tidb/pull/8665

### What is changed and how it works?

There is a little code change in the transformation framework. Group logical property has not derived.

### Check List 

Tests 

 - Unit test
 - Integration test

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8689)

### CODE REVIEW

File: planner/transformation/rule_FilterAggregateTranspose.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import (
+	"github.com/pingcap/tidb/expression"
+	plannercore "github.com/pingcap/tidb/planner/core"
+	"github.com/pingcap/tidb/planner/memo"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/set"
+)
+
+// FilterAggregateTransposeRule pushes the filters through the aggregate:
+// before: filter(aggregate(any), f1, f2, ...)
```
Comment (winoros) [inline planner/transformation/rule_FilterAggregateTranspose.go]: What does `f1`, `f2` mean?

File: planner/transformation/rule_FilterAggregateTranspose.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+// NewFilterAggregateTransposeRule creates a new FilterAggregateTransposeRule.
+func NewFilterAggregateTransposeRule() *FilterAggregateTransposeRule {
+	pattern := memo.BuildPattern(memo.OperandSelection, memo.BuildPattern(memo.OperandAggregation))
+	return &FilterAggregateTransposeRule{baseTransform{pattern}}
+}
+
+// OnTransform push the filters through the aggregate:
+// before: filter(aggregate(any), f1, f2, ...)
```
Comment (winoros) [inline planner/transformation/rule_FilterAggregateTranspose.go]: Need we repeat these comments?
`OnTransform` only keeps `does the actual transformation`.
And leave the detailed comments to where the struct is defined.

File: planner/memo/test_helper.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+package memo
+
+import (
+	"fmt"
+
+	"github.com/pingcap/tidb/planner/core"
+)
+
+// DumpGroupLogicalPlans dumps all the group expressions in the group.
+func DumpGroupLogicalPlans(g *Group) []string {
```
Comment (winoros) [inline planner/memo/test_helper.go]: Is this only used for testing/debugging?

**Comment by zz-jason** [comment]
close due to outdated.

**Pull request closed without merging**